Fix a bug with how ScrollableGrid deals with padding.

mpcomplete opened the pull request:
It wasn't taking padding into account when deciding which children were
visible. I modeled the solution off of the way ScrollableList handles
this.

Fixes https://github.com/flutter/flutter/issues/5522

abarth commented:
Looks good but needs a test.

HansMuller commented on packages/flutter/lib/src/widgets/scrollable_grid.dart:
```diff
@@ -126,16 +126,24 @@ class GridViewport extends VirtualViewportFromIterable {
     assert(delegate != null);
   }
 
+  /// The [startOffset] without taking the [padding] into account.
```
I think you mean the delegate's padding.

mpcomplete committed:
Fix a bug with how ScrollableGrid deals with padding.

It wasn't taking padding into account when deciding which children were
visible. I modeled the solution off of the way ScrollableList handles
this.

Fixes https://github.com/flutter/flutter/issues/5522

mpcomplete committed:
comment

mpcomplete commented:
Added a test.

HansMuller commented:
LGTM. Travis wants some lint love.

mpcomplete committed:
lint
